ultipro/operations.py

```diff
@@ -0,0 +1,363 @@
+        except requests.exceptions.SSLError as e:
+            logger.exception("{}".format(e))
+            raise ConnectorError("{}".format("SSL certificate validation failed"))
+        except requests.exceptions.ConnectionError as e:
+            logger.exception("{}".format(e))
+            raise ConnectorError(
+                "{}".format(
+                    "The request timed out while trying to connect to the remote server"
+                )
+            )
+        except Exception as e:
+            logger.exception("{}".format(e))
+            raise ConnectorError("{}".format(e))
+
+    def __findEmployeePhone(self, pageNumber, pageSize):
+        # This function will perform a generic search to return all people in the company
+        headers = {"Content-Type": "application/soap+xml"}
+        body = f"""<s:Envelope xmlns:a="http://www.w3.org/2005/08/addressing" xmlns:s="http://www.w3.org/2003/05/soap-envelope">
+        <s:Header>
+            <a:Action s:mustUnderstand="1">http://www.ultipro.com/services/employeephoneinformation/IEmployeePhoneInformation/FindPhoneInformations</a:Action>
+            <UltiProToken xmlns="http://www.ultimatesoftware.com/foundation/authentication/ultiprotoken">{self.ultiProToken}</UltiProToken>
+            <ClientAccessKey xmlns="http://www.ultimatesoftware.com/foundation/authentication/clientaccesskey">{self.clientApiKey}</ClientAccessKey>
+        </s:Header>
+        <s:Body>
+            <FindPhoneInformations xmlns="http://www.ultipro.com/services/employeephoneinformation">
+            <query xmlns:b="http://www.ultipro.com/contracts" xmlns:i="http://www.w3.org/2001/XMLSchema-instance">
+                <b:CompanyCode> = {self.companyId}</b:CompanyCode>
+                <b:PageNumber> {pageNumber}</b:PageNumber>
+                <b:PageSize> {pageSize}</b:PageSize>
+            </query>
+            </FindPhoneInformations>
+        </s:Body>
+        </s:Envelope>"""
+        try:
+            response = requests.post(
+                "https://service5.ultipro.com/services/EmployeePhoneInformation",
+                data=body,
+                headers=headers,
+            )
+            if response.ok:
+                response_dict = xmltodict.parse(response.text)
+                return response_dict
+            else:
+                raise ConnectorError(response.text)
+        except requests.exceptions.ConnectionError as e:
+            logger.exception("{}".format(e))
+            raise ConnectorError(
+                "{}".format(
+                    "The request timed out while trying to connect to the remote server"
+                )
+            )
+        except Exception as e:
+            logger.exception("{}".format(e))
+            raise ConnectorError("{}".format(e))
+
+    def _findEmployeePhone(self, params):
+        # iterate through all pages and return employeeList
+        pageNumber = params.get("pageNumber", "all")
```

Comment:
if pageNumber value is expected to be "all" then please specify it in the tooltip.